tests/Unit/Filesystem/ComposerReaderTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the TYPO3 project  - inspiring people to share!
+ * (c) 2020 Oliver Bartsch & Benni Mack
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace TYPO3\Tailor\Tests\Unit\Filesystem;
+
+use PHPUnit\Framework\TestCase;
+use TYPO3\Tailor\Filesystem\ComposerReader;
+
+class ComposerReaderTest extends TestCase
+{
+    private const COMPOSER_FILE = 'tmp/composer.json';
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        mkdir('tmp');
+    }
+
+    protected function tearDown(): void
+    {
+        unlink(self::COMPOSER_FILE);
+        rmdir('tmp');
+        parent::tearDown();
+    }
+
+    /**
+     * @test
+     */
+    public function throwsExceptionOnEmptyComposerJsonFile(): void
+    {
+        $this->expectExceptionCode(1610442954);
```

Comment:
whenever possible, please have both ->expectExceptionCode() and ->expectException()

Reply:
Done